src/Runtime/Distributions/SequenceDistribution.cs

```diff
@@ -261,6 +261,28 @@ public static TThis Concatenate(IEnumerable<TElementDistribution> sequence)
             return FromWorkspace(result.GetAutomaton());
         }
 
+
+        /// <summary>
+        /// Creates a distribution over sequences induced by a given list of distributions over sequence elements
+        /// where the sequence can optionally end at any length. The last distribution in the list is used for a self-loop,
+        /// allowing any length sequence.
+        /// </summary>
+        /// <param name="elementDistributions">Enumerable of distributions over sequence elements and the transition weights.</param>
+        /// <returns>The created distribution.</returns>
+        public static TThis WordModel(IList<TElementDistribution> elementDistributions)
```

Comment:
I would suggest calling this something like ElementSequence and having a bool to say whether the last element should be repeated.

Reply:
I have extended the Concatenate signature to optionally do this case. Default argument settings are such that existing code that uses Concatenate does not need to change.